Sprint 5 UI tests

Yulia7052 opened the pull request.

Yulia7052 committed:
Sprint 5 UI tests

k-asadova commented on consts.py:
```diff
@@ -0,0 +1,26 @@
+site_url = 'https://stellarburgers.nomoreparties.site/'
```
Константы все обозначаем капслоком

k-asadova commented on consts.py:
```diff
@@ -0,0 +1,26 @@
+login_registration_button = '//*[@id="root"]/div/main/div/div/p[1]/a'
+enter_button = '//*[@id="root"]/div/main/div/div/p/a'
+registration_submit_button = '//*[@id="root"]/div/main/div/form/button'
+login_enter_button = '//*[@id="root"]/div/main/div/form/button'
```
Сейчас почти все локаторы сильно завязаны на структуру страницы - это делает их нестабильными. Попробуй привязываться не по иерархии, а по атрибутам элемента - классу, тексту, итд.

k-asadova commented on tests/test_constructor_open_filling_tab.py:
```diff
@@ -0,0 +1,15 @@
+    tab_select_class_attr = filling_div.get_attribute('class')
+
+    assert 'tab_tab_type_current' in tab_select_class_attr
+    driver.quit()
```
driver.quit() нужно в фикстуру после yield

k-asadova commented on tests/test_enter_profile.py:
```diff
@@ -0,0 +1,25 @@
+    driver.find_element(By.XPATH, consts.header_profile_button).click()
+
+    WebDriverWait(driver, 3).until(expected_conditions.url_to_be('https://stellarburgers.nomoreparties.site/account/profile'))
+    assert driver.current_url == 'https://stellarburgers.nomoreparties.site/account/profile'
```
URL в константы, стараемся не допускать хардкода в тестах

k-asadova commented on tests/test_enter_profile.py:
```diff
@@ -0,0 +1,25 @@
+    driver.find_element(By.XPATH, consts.constructor_reserve_button).click()
+
+    WebDriverWait(driver, 3).until(expected_conditions.visibility_of_element_located((By.XPATH, consts.login_email_field)))
+    email = 'ulia_manaenkova_10_088@google.com'
```
Детали пользователя тоже выносим в константы либо - лучше - создаем метод, который нам их генерирует

k-asadova commented on tests/test_exit_profile.py:
```diff
@@ -0,0 +1,29 @@
+    driver.find_element(By.XPATH, consts.constructor_reserve_button).click()
+
+    WebDriverWait(driver, 3).until(expected_conditions.visibility_of_element_located((By.XPATH, consts.login_email_field)))
+    email = 'ulia_manaenkova_10_088@google.com'
```
Тут аналогично

k-asadova commented on tests/test_constructor_open_bun_tab.py:
```diff
@@ -0,0 +1,18 @@
+from selenium.webdriver.chrome.service import Service
```
Неиспользуемый импорт во всех тестах

klushcheva reviewed (commented):
Юлия, привет! Тесты проходят, но нужно поработать над деталями реализации. Буду ждать твоих правок

Yulia7052 committed:
Sprint 5 Modify fixture

Yulia7052 committed:
Sprint 5 Updates

- Remove unnecessary import
- Modify caps consts
- Modify locators
- Remove hardcode from tests

Yulia7052 committed:
Sprint 5 Readme update

k-asadova commented on consts.py:
```diff
@@ -1,26 +1,36 @@
-login_recovery_password_button = '//*[@id="root"]/div/main/div/div/p[2]/a'
-account_recovery_exit_button = '//*[@id="root"]/div/main/div/nav/ul/li[3]/button'
-constructor_enter_button = '//*[@id="root"]/div/header/nav/ul/li[1]/a'
+HEADER_PROFILE_BUTTON = '//*[@id="root"]/div/header/nav/a'
```
Нужно поправить: все еще завязанные на структуру страницы локаторы

k-asadova commented on consts.py:
```diff
@@ -1,26 +1,36 @@
-constructor_bun_div = '//*[@id="root"]/div/main/section[1]/div[1]/div[1]'
-constructor_filling_div = '//*[@id="root"]/div/main/section[1]/div[1]/div[3]'
+REGISTRATION_PASS_ERROR_CLASS = 'input_status_error'
+BURGER_DIVS_CLASS = 'tab_tab__1SPyG'
```
Вот тут нужно искать по xpath и class @contains tab_tab. Эта цифробуквенная часть 1SPyG сгенерирована автоматически при деплое. При последующих апдейтах локатор может сломаться

k-asadova commented on tests/conftest.py:
```diff
@@ -3,4 +3,6 @@
-    return webdriver.Chrome()
+    driver = webdriver.Chrome()
+    yield driver
+    driver.quit()
```
Хорошо!

klushcheva reviewed (commented):
Юлия, хорошо - осталось только поправить локаторы и будет отлично!

Yulia7052 committed:
Sprint 5 modify consts
